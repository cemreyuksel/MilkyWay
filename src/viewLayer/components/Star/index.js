import React from 'react';

import { Card } from 'react-bootstrap';

import { useFetch } from '../../hooks';
import { UNIVERSE_URL, LOADING_MESSAGE } from '../../../dataLayer/env';

const cardMap = {
  RED: 'danger',
  BLUE: 'primary',
  GREEN: 'success',
  YELLOW: 'warning',
  BLACK: 'dark',
};

const getUniverseName = (universes, universeId) => {
  const { name } = universes.find((universe) => universe.id === universeId);
  return name;
};

const Star = ({ name, color, universeId }) => {
  const [data, loading] = useFetch(UNIVERSE_URL);

  return (
    <>
      {loading ? (
        <h2>{`${LOADING_MESSAGE}`}</h2>
      ) : (
        <Card
          bg={`${cardMap[color]}`}
          style={{ width: '20rem' }}
          className="text-center"
        >
          <Card.Body>
            <Card.Title> {name}</Card.Title>
            <Card.Header>
              Universe: {getUniverseName(data, universeId)}
            </Card.Header>
          </Card.Body>
        </Card>
      )}
    </>
  );
};

export default Star;