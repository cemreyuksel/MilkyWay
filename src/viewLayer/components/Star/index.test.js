import React from 'react';
import TestRenderer from 'react-test-renderer';
import Star from './index';

jest.mock('../../hooks', () => ({
  useFetch: () => [
    [
      {
        id: 333,
        maxSize: 3,
        name: 'Big universe',
      },
    ],
    false,
  ],
}));

const mockName = 'some star name';
const mockColor = 'red';
const mockUniverseId = 333;

describe('Stars component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const tree = TestRenderer.create(
      <Star name={mockName} mockColor={mockColor} universeId={mockUniverseId} />
    ).toJSON();;

    expect(tree).toMatchSnapshot();
  });

  test('displays the corresponding name of the universe', () => {
    const expectedChildren = ['Universe: ', 'Big universe']
    const renderer = TestRenderer.create(
      <Star name={mockName} mockColor={mockColor} universeId={mockUniverseId} />
    );
    const instance = renderer.root;
    
    expect(instance.findByProps({className: "card-header"}).children).toEqual(expectedChildren);
  });
});
