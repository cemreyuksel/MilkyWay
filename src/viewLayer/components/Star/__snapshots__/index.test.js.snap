// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars component renders 1`] = `
<div
  className="text-center card bg-undefined"
  style={
    Object {
      "width": "20rem",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
       
      some star name
    </div>
    <div
      className="card-header"
    >
      Universe: 
      Big universe
    </div>
  </div>
</div>
`;
